apis/kafkamanagement/v1beta1/kafkauser_types.go

```diff
@@ -36,6 +36,7 @@ type KafkaUserSpec struct {
 type KafkaUserOptions struct {
 	OverrideExistingUser bool   `json:"overrideExistingUser,omitempty"`
 	SASLSCRAMMechanism   string `json:"saslScramMechanism"`
+	AuthMechanism        string `json:"authMechanism"`
 }
```

Comment:
Please don't use `KafkaUserOptions`. Just extract these fields into `Spec`. There are good reasons for that:
1. Compliance with the Kafka User model of the Instaclustr API.
2. It will make Option not required.

Reply:
Agree, fixed